tools/prefetchlistStoreService/main.go

```diff
@@ -0,0 +1,96 @@
+/*
+* Copyright (c) 2023. Nydus Developers. All rights reserved.
+*
+* SPDX-License-Identifier: Apache-2.0
+ */
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+)
+
+type PrefetchFile struct {
+	Path string
+}
+
+type CacheItem struct {
+	ImageName     string
+	ContainerName string
+	PrefetchFiles []PrefetchFile
+}
+
+type Cache struct {
+	Items map[string]*CacheItem
+	mutex sync.Mutex
+}
+
+func (cache *Cache) key(imageName, containerName string) string {
+	return fmt.Sprintf("%s,%s", imageName, containerName)
+}
+
+func (cache *Cache) Get(imageName, containerName string) *CacheItem {
+	cache.mutex.Lock()
+	defer cache.mutex.Unlock()
+
+	return cache.Items[cache.key(imageName, containerName)]
+}
+
+func (cache *Cache) Set(item CacheItem) {
+	cache.mutex.Lock()
+	defer cache.mutex.Unlock()
+
+	cache.Items[cache.key(item.ImageName, item.ContainerName)] = &item
+}
+
+var serverCache Cache
+
+func uploadHandler(w http.ResponseWriter, r *http.Request) {
+	var item CacheItem
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, "Failed to read request body", http.StatusBadRequest)
+		return
+	}
+	err = json.Unmarshal(body, &item)
+	if err != nil {
+		http.Error(w, "Invalid request payload", http.StatusBadRequest)
+		return
+	}
+
+	serverCache.Set(item)
+	fmt.Fprintf(w, "Uploaded CacheItem for %s, %s successfully", item.ImageName, item.ContainerName)
+}
+
+func downloadHandler(w http.ResponseWriter, r *http.Request) {
+	imageName := r.URL.Query().Get("imageName")
+	containerName := r.URL.Query().Get("containerName")
+
+	item := serverCache.Get(imageName, containerName)
+	if item == nil {
+		http.Error(w, "CacheItem not found", http.StatusNotFound)
+		return
+	}
+
+	err := json.NewEncoder(w).Encode(item)
+	if err != nil {
+		http.Error(w, "Failed to encode CacheItem to JSON", http.StatusInternalServerError)
+		return
+	}
+}
+
+func main() {
+	serverCache = Cache{Items: make(map[string]*CacheItem)}
+
+	http.HandleFunc("/api/v1/prefetch/upload", uploadHandler)
```

Comment:
Should only accept POST for `upload` and GET for `download`, we'd better use the `echo` web framework.